Cache/IO.lean

```diff
@@ -263,45 +259,45 @@ partial def getFilesWithExtension
     (← fp.readDir).foldlM (fun acc dir => getFilesWithExtension dir.path extension acc) acc
   else return if fp.extension == some extension then acc.push fp else acc
 
-abbrev HashMap := Std.HashMap FilePath UInt64
+/--
+The Hash map of the cache.
+-/
+abbrev NameHashMap := Std.HashMap FilePath UInt64
```

Comment:
Suggested name in review of `#21238`. This PR will change it to `Std.HashMap Name UInt64`, thus the name.